haystack/preview/document_stores/protocols.py

```diff
@@ -0,0 +1,114 @@
+from typing import Protocol, Optional, Dict, Any, List, Literal
+
+import logging
+
+from haystack.preview.dataclasses import Document
+
+
+logger = logging.getLogger(__name__)
+DuplicatePolicy = Literal["skip", "overwrite", "fail"]
```

Comment:
I would make this an `Enum`, it would make `write_documents` calls clearer.

With a literal we'd call it like so:

```
store.write_documents(docs,  "fail")
```

The `fail` string might not be clear what it refers. Instead if we make `DuplicatePolicy` an `enum` we could have something like this:

```
store.write_documents(docs,  DuplicatePolicy.Fail)
```

It's clear right away that it refers to the duplicate policy, and also safer to use I think.

Reply:
Good idea!